android/app/src/main/java/updated/mysterium/vpn/ui/base/BaseActivity.kt

```diff
@@ -220,20 +220,21 @@ abstract class BaseActivity : AppCompatActivity() {
             } else {
                 unprotectedConnection()
             }
-        })
-        baseViewModel.insufficientFunds.observe(this, {
+        }
+        baseViewModel.insufficientFunds.observe(this) {
             insufficientFundsPopUp()
-        })
-        baseViewModel.isInternetNotAvailable.observe(this, { isAvailable ->
-            isInternetAvailable = isAvailable
+        }
+        baseViewModel.isInternetAvailable.observe(this) { isAvailable ->
             if (!isAvailable) {
+                isInternetAvailable = isAvailable
                 wifiNetworkErrorPopUp()
-            } else {
+            } else if (!isInternetAvailable) {
+                isInternetAvailable = isAvailable
```

Comment:
`isInternetAvailable = isAvailable` this line of code using in both statement of if-else, so we can move out this line of code on top of checkState. 
```
isInternetAvailable = isAvailable

if (isInternetAvailable.not()) {
   wifiNetworkErrorPopUp()
} else {
   wifiErrorDialog?.dismiss()
   wifiErrorDialog = null
   retryLoading()
}
```

Reply:
Done